File: 001-010/004/004.py
#!/usr/bin/env python

# Find the largest palindrome made from the product of two 3-digit numbers.


def isPalindrome(k):
    for s in range(0, len(k)//2):
        if k[s] != k[-(1+s)]:
            return False
    return True


def main():
    total = []
    for i in range(999, 99, -1):
        for j in range(999, 99, -1):
            k = str(i * j)
            if isPalindrome(k) is True:
                total.append(int(k))
    print("The largest palindrome in question is", max(total))


main()

File: 001-010/004/004.php
<?php

function isPalindrome($k) {
    for ($s = 0; $s < strlen($k) / 2; $s++) {
        if (substr($k, $s, 1) != substr($k, -(1+$s), 1)) {
            return False;
        }
    }
    return True;
}



$max = 0;
for($i = 999; $i > 99; $i--) {
    for($j = 999; $j > 99; $j--) {
        $k = $i * $j;
        if (isPalindrome($k) == True) {
            if ($k > $max) {
                $max = $k;
            }
        }
    }
}

echo "The largest palindrome in question is " . $max . ".\n"

?>
